modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeImportsCheck.java

```diff
@@ -0,0 +1,177 @@
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.util.ArrayUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.source.formatter.JSPImportsFormatter;
+import com.liferay.source.formatter.parser.JavaClass;
+import com.liferay.source.formatter.parser.JavaClassParser;
+import com.liferay.source.formatter.util.SourceFormatterUtil;
+
+import java.io.InputStream;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Hugo Huijser
+ * @author Nícolas Moura
+ */
+public class UpgradeImportsCheck extends BaseFileCheck {
+
+	@Override
+	protected String doProcess(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		_getImportsMap();
+
+		if (!fileName.endsWith(".java") && !fileName.endsWith(".jsp") &&
+			!fileName.endsWith(".ftl")) {
+
+			return content;
+		}
+
+		return _fixImports(fileName, content);
+	}
+
+	private static String _replaceVariables(String content) {
+		String newContent = content;
+
+		if (!_variablesMap.isEmpty()) {
+			newContent = StringUtil.replace(
+				content, ArrayUtil.toStringArray(_variablesMap.keySet()),
+				ArrayUtil.toStringArray(_variablesMap.values()), true);
+
+			for (Map.Entry<String, String> entry : _variablesMap.entrySet()) {
+				String regex = StringBundler.concat(
+					"\\b([_a-z]\\w*)", entry.getKey(), "\\b");
+
+				Pattern pattern = Pattern.compile(regex);
+
+				Matcher variableMatcher = pattern.matcher(newContent);
+
+				if (variableMatcher.find()) {
+					newContent = newContent.replaceAll(
+						regex, variableMatcher.group(1) + entry.getValue());
+				}
+			}
+		}
+
+		return newContent;
+	}
+
+	private void _checkClassNames() {
+		for (Map.Entry<String, String> entry : _importsMap.entrySet()) {
+			String className = SourceFormatterUtil.getSimpleName(
+				entry.getKey());
+			String newClassName = SourceFormatterUtil.getSimpleName(
+				entry.getValue());
+
+			if (!className.equals(newClassName)) {
+				_variablesMap.put(className, newClassName);
+				_variablesMap.put(
+					StringUtil.lowerCaseFirstLetter(className),
+					StringUtil.lowerCaseFirstLetter(newClassName));
+			}
+		}
+	}
+
+	private String _fixImports(String fileName, String content)
+		throws Exception {
+
+		List<String> importNames = new ArrayList<>();
+
+		String newContent = content;
+
+		if (fileName.endsWith(".java")) {
+			JavaClass javaClass = JavaClassParser.parseJavaClass(
+				fileName, content);
+
+			importNames = javaClass.getImportNames();
+
+			newContent = javaClass.getContent();
+		}
+		else if (fileName.endsWith(".jsp")) {
+			importNames = JSPImportsFormatter.getImportNames(content);
+		}
+		else {
+			Matcher matcher = _ftlImportNamePattern.matcher(content);
+
+			while (matcher.find()) {
+				importNames.add(matcher.group(1));
+			}
+		}
+
+		_variablesMap = new HashMap<>();
```

Comment:
Can `_variablesMap` be a local variable that we pass as a parameter to `_checkClassNames` and `_replaceVariables` ?